check-dependencies.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+set -euo pipefail
+
+# Testing with this to simulate different installed apps:
+# docker run -it -v ~/Source/aws-cdk:/var/cdk ubuntu bash
+
+# Note: Don't use \d in regex, it doesn't work on Macs without GNU tools installed
+# Use [0-9] instead
+
+app=""
+app_min=""
+app_v=""
+
+die() { echo "$*" 1>&2 ; exit 1; }
+wrong_version() { echo "Found $app version $app_v. Install $app >= $app_min" 1>&2; exit 1; }
+
+check_which() {
+    app=$1
+    min=$2
+
+    echo -e "Checking if $app is installed... \c"
+
+    w=$(which ${app}) || w=""
+
+    if [ -z $w ] || [ $w == "$app not found" ]
+    then
+        die "Missing dependency: $app. Install $app >= $min"
+    else
+        echo "Ok"
+    fi
+}
+
+# [Node.js >= 10.13.0]
+#   ⚠️ versions `13.0.0` to `13.6.0` are not supported due to compatibility issues with our dependencies.
+app="node"
+app_min="v10.13.0"
```

Comment:
why do we need to store these into global variables?

This feels way more readable to me:

```bash
check_which node v10.13.0
```

Reply:
The wrong_version function relies on them.